spring-messaging/src/main/java/org/springframework/messaging/converter/ProtobufMessageConverter.java

```diff
@@ -227,9 +225,25 @@ protected Object convertToInternal(Object payload, @Nullable MessageHeaders head
 		return payload;
 	}
 
-	@Override
-	protected boolean supports(Class<?> clazz) {
-		return Message.class.isAssignableFrom(clazz);
+
+	/**
+	 * Create a new {@code Message.Builder} instance for the given class.
+	 * <p>This method uses a ConcurrentReferenceHashMap for caching method lookups.
+	 */
+	private Message.Builder getMessageBuilder(Class<?> clazz) {
+		try {
+			assert supports(clazz);
+
+			Method method = methodCache.get(clazz);
+			if (method == null) {
+				method = clazz.getMethod("newBuilder");
+				methodCache.put(clazz, method);
+			}
+			return (Message.Builder) method.invoke(clazz);
+		} catch (Exception ex) {
```

Comment:
Correct the braces throughout, please :)

Reply:
Sorry. it is my first contribution to spring. thanks for correcting me